src/Lecture3.hs

```diff
@@ -137,7 +180,8 @@ data List1 a = List1 a [a]
 
 -- | This should be list append.
 instance Semigroup (List1 a) where
-
+  (<>) :: List1 a -> List1 a -> List1 a
+  (List1 x xs) <> (List1 y ys) = List1 x (xs <> [y] <> ys)
```

Comment:
You can slightly improve the performance here by using the `:` operator instead of list append `<>` or `++` 🐎 
```suggestion
  List1 x xs <> List1 y ys = List1 x (xs <> (y : ys))
```

Reply:
How would I know (:) is implemented ? Does Semigroup also defined this constructor / function ?

Reply:
`:` is an operator of type:

```haskell
(:) :: a -> [a] -> [a]
```

It's always available and it works for lists. So you can always use it 🙂